base/scripts/code.player.cs

```diff
@@ -244,35 +246,47 @@
         }
         
         //handle damage events here - ignoring all team dealt damage / self damage
-        if (%teamDamageOccured == 0) {
-            
+        if ($Collector::DamageEnabled && %teamDamageOccured == 0) {
             
+            if(%shooterClient == %damagedClient) { return; }
+
             %newValue = (%value * 150);
-            %newRemainder = (%newValue - floor(%newValue));
             
-            if (%newRemainder >= 0.5) {
-                %newValue = floor((%newValue + 1));
+            if (%spillOver < 0) {
+                //do nothing
             }
             else {
-                %newValue = floor(%newValue);
+                %newValue = (%newValue - (%spillOver * 150));
             }
             
+            $DmgTracker::DamageOut[%shooterClient] += %newValue;
+            $DmgTracker::DamageIn[%damagedClient] += %newValue;
+              
+        }
+        
+        else if ($Collector::DamageEnabled && %teamDamageOccured == 1) {
+            
+            %newValue = (%value * 150);
+            
             if (%spillOver < 0) {
-                //MessageAll(0, "Damage Dealt: " @ %newValue);
+                //do nothing
             }
             else {
-                %newValue = (1+(%newValue - floor(%spillOver * 150)));
-                //MessageAll(0, "New Damage Dealt: " @ %newValue);
+                %newValue = (%newValue - (%spillOver * 150));
             }
             
-            if ($Collector::DamageEnabled) {
-                
-                $DmgTracker::DamageOut[%shooterClient] += %newValue;
-                $DmgTracker::DamageIn[%damagedClient] += %newValue;
-                
-                //zadmin::ActiveMessage::All( DamageDealt, %shooterClient, %damagedClient, %newValue );
-            }
+            $DmgTracker::TeamDamageOut[%shooterClient] += %newValue;
+            $DmgTracker::TeamDamageIn[%damagedClient] += %newValue;
+
         }
+        else { }
+    }
+    
+    if ( (%type == $BulletDamageType) && (%teamDamageOccured == 0) ) {

```

Comment:
i'd like to call out that this is a change in behavior - the server used to send the hit sound for friendly fire damage but the new code doesn't. are you sure that's what people want?

Reply:
originally, hit sound was for enemies only, you can check in roscos, when we switched to cg hit sounds only... (stork made it so hit sounds for cg were for both enemy and friendly this change was never discussed just shoehorned in)... So essentially this is going back to how things were.

Reply:
makes sense - ty!